shell/edit/provisioning.cattle.io.cluster/__tests__/rke2.test.ts

```diff
@@ -293,4 +214,57 @@ describe('component: rke2', () => {
 
     expect((select.vm as unknown as any).disabled).toBe(disabled);
   });
+
+  it.each([
+    ['custom', true],
+    ['anything else', false] // without proper data, machine pool is always not present
+  ])('should allow creation of RKE2 cluster with provider %p if pool machines are missing (%p)', (provider, result) => {
+    const k8s = 'v1.25.0+rke2r1';
+    const wrapper = mount(rke2, {
+      propsData: {
+        mode:  'create',
+        value: {
+          spec: {
+            ...defaultSpec,
+            kubernetesVersion: k8s,
+
+          }
+        },
+        provider,
+      },
+      computed: defaultComputed,
+      mocks:    {
+        ...defaultMocks,
+        $store: { getters: defaultGetters },
+      },
+      stubs: defaultStubs
+    });
+
+    expect(wrapper.vm.validationPassed()).toBe(result);
+  });
+
+  it('should allow creation of K3 clusters if pool machines are missing', () => {
```

Comment:
Just for my understanding of the functional requirements, what is the difference between rke2 and k3s validations ? If the k8s version makes any difference, we are duplicating this test case.

Reply:
RKE2 requires the node pool machines, the K3 does not.

Theoretically we are preventing in the first case (if it's a provided cluster) and allowing in the second one.